pubsub/kafka/kafka.go

```diff
@@ -308,6 +335,33 @@ func (k *Kafka) Subscribe(req pubsub.SubscribeRequest, handler pubsub.Handler) e
 	return nil
 }
 
+// upgradeMetadata updates metadata properties based on deprecated usage.
+func (k *Kafka) upgradeMetadata(metadata pubsub.Metadata) (pubsub.Metadata, error) {
+	// If authType is not set, derive it from authRequired.
+	if val, ok := metadata.Properties["authType"]; !ok || val == "" {
```

Comment:
How about get those value first then check to reduce `if` block?

Reply:
I've refactored this to reduce the nested levels.